### PR TITLE
Restructure the shape of the guide and focus on formatting vs linting

### PR DESCRIPTION
**Pull request by whatyouhide**
We will lose only a couple links (`#source-code-layout`) but it's a price that I am very willing to pay (it will still link to the guide, just not the proper section).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -300,53 +343,85 @@
...
+
+  prefix =
+    case base do
+      :binary -> "0b"
+      :octal -> "0o"
+      :hex -> "0x"
+    end
   ```
 
 * <a name="binary-operators-at-eols"></a>
```
Comment (lexmag) [inline README.md:433]: Perhaps we should mention one exception to the rule:
```elixir
message =
  "No matching message.\n" <>
  "Process mailbox:\n" <>
  mailbox
```
WDYT?

**Comment by lexmag** [comment]
Shall we quote this comment somewhere: https://github.com/elixir-lang/elixir/pull/6022#discussion_r112669015? :bowtie:

File: README.md (hunk 1 above)
Comment (whatyouhide) [reply README.md:433]: Not sure I see the exception there, is it the `=` operator?

**Comment by whatyouhide** [comment]
> Shall we quote this comment somewhere: elixir-lang/elixir#6022 (comment)? :bowtie:

Added a quote at the beginning of the guide.

File: README.md (hunk 1 above)
Comment (lexmag) [reply README.md:433]: When we have a binding or a continuation of the argument construction we don't indent next lines to one more level.

File: README.md [hunk 2]
```diff
@@ -4,19 +4,42 @@
 >
 > — <cite>What a Programmer Does, 1967</cite>
 
+#### Why follow this guide?
+
+> The only tool available right now is this guide written using the blood of the maintainers as ink
```
Comment (lexmag) [inline README.md]: Period?

**Comment by whatyouhide** [comment]
@lexmag addressed feedback, ready to merge.

**Review by lexmag — APPROVED** [review]
💛

**Pull request merged**